src/saml2/config.py

```diff
@@ -569,13 +585,198 @@ def ecp_endpoint(self, ipaddress):
+    def get_endpoint_element(self, element):
+        return getattr(self, "_sp_endpoints", {}).get(element, None)
+
+    def get_application_identifier(self):
+        return getattr(self, "_sp_application_identifier", None)
+
+    def get_protocol_version(self):
+        return getattr(self, "_sp_protocol_version", None)
+
+    def get_node_country(self):
+        return getattr(self, "_sp_node_country", None)
+
+    @property
+    def warning_validators(self):
+        sp_warning_validators = {
+            "hide_assertion_consumer_service SHOULD be set to True":
+                getattr(self, "_sp_hide_assertion_consumer_service", None) is True
+        }
+        return {**super().warning_validators, **sp_warning_validators}
+
+    @property
+    def error_validators(self):
+        sp_error_validators = {
+            "authn_requests_signed MUST be set to True":
+                getattr(self, "_sp_authn_requests_signed", None) is True,
+            "sp_type MUST be set to 'public' or 'private'":
+                getattr(self, "_sp_sp_type", None) in ("public", "private")
+        }
+        return {**super().error_validators, **sp_error_validators}
+
+
 class IdPConfig(Config):
     def_context = "idp"
 
     def __init__(self):
         Config.__init__(self)
 
 
+class eIDASIdPConfig(IdPConfig, eIDASConfig):
+    def get_endpoint_element(self, element):
+        return getattr(self, "_idp_endpoints", {}).get(element, None)
+
+    def get_application_identifier(self):
+        return getattr(self, "_idp_application_identifier", None)
+
+    def get_protocol_version(self):
+        return getattr(self, "_idp_protocol_version", None)
+
+    def get_node_country(self):
+        return getattr(self, "_idp_node_country", None)
+
+    def verify_non_notified_loa(self):
+        return not any(
+            [x.startswith("http://eidas.europa.eu/LoA/")
+             for x in getattr(self, "_idp_supported_loa", {}).get("non_notified")]
+        )
+
+    def verify_notified_loa(self):
+        return all(
+            [x in ["http://eidas.europa.eu/LoA/low",
+                   "http://eidas.europa.eu/LoA/substantial",
+                   "http://eidas.europa.eu/LoA/high"]
+             for x in getattr(self, "_idp_supported_loa", {}).get("notified")]
+        )
+
+    @property
+    def warning_validators(self):
+        idp_warning_validators = {}
+        return {**super().warning_validators, **idp_warning_validators}
+
+    @property
+    def error_validators(self):
+        idp_error_validators = {
+            "want_authn_requests_signed MUST be set to True":
+            getattr(self, "_idp_want_authn_requests_signed", None) is True,
+            "provided_attributes MUST be set to denote the supported attributes by "
+            "the IdP":
+            not_empty(getattr(self, "_idp_provided_attributes", None)),
+            "supported_loa for non-notified eIDs MUST NOT use an "
+            "http://eidas.europa.eu/LoA/ prefix":
```

Comment:
Is this true? Does a spec mandate that I cannot use the same prefix?

Reply:
Yeap

> To support reuse of eIDAS-Node infrastructure for non-notified eID schemes, Member States MAY support other URIs. Such non-notified eIDs MUST NOT use an “http://eidas.europa.eu/LoA/” prefix.

Section 3.2 of eIDAS SAML Message Format v.1.2 Final

Reply:
That is scoped only to non-notified eIDs. Only eIDs that are non-notified should error out if this happens.

Reply:
Not sure what you mean. The "IdP" part may optionally support non-notified eIDs. The spec says that this must be published (which eIDs are supported) and it states a rule these eIDs must follow (about the prefix). So the eIDASIdPConfig cannot support (aka contain them in the configuration) invalid eIDs

Reply:
OK, I thought the validator was iterating over all LoAs, but it does something else:

```python
getattr(self, "_idp_supported_loa", {}).get("notified") # or "non_notified"
```

What is `_idp_supported_loa`?

Reply:
these are all the supported LoA for the IdP. It contains the LoAs for notified eIDs (the "officially" supported) and the LoAs for the non-notified eIDs. It's not mandatory for an IdP to support all three official LoAs, they can choose only to support `http://eidas.europa.eu/LoA/high` for example

P.S: Obviously there's documentation missing from these changes. But I figured we settle with the code first and then write the documentation

Reply:
is it? https://github.com/IdentityPython/pysaml2/pull/658/files#diff-d6a3728fa3e2edb048e6e56a35520d84R133

Reply:
That is `supported_loa` not `_idp_supported_loa`. You are invoking into the internals of something that you should not.